norddrop/src/device/mod.rs

```diff
@@ -593,36 +594,79 @@ fn open_database(
     }
 
-    // Still problems? Let's try to delete the file
-    warn!(logger, "Removing old DB file");
-    if let Err(err) = std::fs::remove_file(dbpath) {
-        error!(
-            logger,
-            "Failed to opend DB and failed to remove it's file: {err}"
+    // If we can't even open the DB in memory, there is nothing else left to do,
+    // throw an error
+    if dbpath == ":memory:" {
+        let error = ffi::types::NORDDROP_RES_DB_ERROR;
+        let error_msg = "Failed to open in-memory DB";
+        moose.developer_exception(
+            -1,
+            error as i32,
+            "".to_string(),
+            error_msg.to_string(),
+            "Database Error".to_string(),
         );
-        return Err(ffi::types::NORDDROP_RES_DB_ERROR);
+
+        error!(logger, "{}", error_msg);
+        Err(error)
     } else {
-        // Inform app that we wiped the old DB file
-        events.dispatch(types::Event::RuntimeError {
-            status: drop_core::Status::DbLost,
-        });
-    };
+        moose.developer_exception(
+            -1,
+            ffi::types::NORDDROP_RES_DB_ERROR as i32,
+            "Initial DB open failed, recreating".to_string(),
+            "Failed to open database".to_string(),
+            "Database Error".to_string(),
+        );
+        // Still problems? Let's try to delete the file, provided it's not in memory
+        warn!(logger, "Removing old DB file");
+        if let Err(err) = std::fs::remove_file(dbpath) {
+            let error_msg = format!("Failed to open DB and failed to remove it's file: {err}");
+            moose.developer_exception(
+                -1,
+                ffi::types::NORDDROP_RES_DB_ERROR as i32,
+                "".to_string(),
+                error_msg.to_string(),
+                "Database Error".to_string(),
+            );
+            error!(logger, "{}", error_msg);
+            // Try to at least open db in memory if the path doesn't work
+            return open_database(":memory:", events, logger, moose);
+        } else {
+            // Inform app that we wiped the old DB file
+            events.dispatch(types::Event::RuntimeError {
+                status: drop_core::Status::DbLost,
+            });
+        };
 
-    // Final try after cleaning up old DB file
-    match drop_storage::Storage::new(logger.clone(), dbpath) {
-        Ok(storage) => Ok(storage),
-        Err(err) => {
-            error!(logger, "Failed to prepare storage: {err}");
-            Err(ffi::types::NORDDROP_RES_DB_ERROR)
+        // Final try after cleaning up old DB file
+        match drop_storage::Storage::new(logger.clone(), dbpath) {
+            Ok(storage) => Ok(storage),
+            Err(err) => {
+                let error = ffi::types::NORDDROP_RES_DB_ERROR;
+                let error_msg = format!("Failed to open DB after cleaning up old file: {err}");
+                moose.developer_exception(
+                    -1,
+                    error as i32,
```

Comment:
Isn't moose using the status codes? I don't know if it even matters but this one is just a cast of the ffi error, not a status

Reply:
This is a little different than the other moose calls that we have, as it's generic for all of the trackers, so we use the argument that seems to fit best

Reply:
Ok, I guess it's not very important after all since we have the description passed into the function